src/main/java/fi/aalto/cs/apluscourses/model/ModuleVirtualFileVisitor.java

```diff
@@ -0,0 +1,34 @@
+package fi.aalto.cs.apluscourses.model;
+
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.openapi.vfs.VirtualFileVisitor;
+import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import java.time.ZonedDateTime;
+import org.jetbrains.annotations.NotNull;
+
+public class ModuleVirtualFileVisitor extends VirtualFileVisitor<Object> {
+
+  private volatile boolean hasChanges = false;
+  private final long downloadedAt;
+
+  public boolean isHasChanges() {
```

Comment:
funny name :smile:

Reply:
autogenerated by IJ, I'll just change it to `hasChanges()` as it reads better this way, despite overlaps with the member name :yum: